src/SFA.DAS.PR.Application/AccountProviders/Queries/GetAccountProviders/GetAccountProvidersQueryHandler.cs

```diff
@@ -0,0 +1,25 @@
+﻿using MediatR;
+using SFA.DAS.PR.Application.Mediatr.Responses;
+using SFA.DAS.PR.Domain.Entities;
+using SFA.DAS.PR.Domain.Interfaces;
+
+namespace SFA.DAS.PR.Application.AccountProviders.Queries.GetAccountProviders;
+
+public class GetAccountProvidersQueryHandler(IAccountProvidersReadRepository accountProvidersReadRepository) : IRequestHandler<GetAccountProvidersQuery, ValidatedResponse<GetAccountProvidersQueryResult>>
+{
+    private readonly IAccountProvidersReadRepository _accountProvidersReadRepository = accountProvidersReadRepository;
+
+    public async Task<ValidatedResponse<GetAccountProvidersQueryResult>> Handle(GetAccountProvidersQuery query, CancellationToken cancellationToken)
+    {
+        List<AccountProvider> accountProviders = await _accountProvidersReadRepository.GetAccountProviders(query.AccountId, cancellationToken);
+
+        if (!accountProviders.Any())
```

Comment:
This could actually have impact on existing consumers as I cannot see a NotFound response being returned. While this is correct, to reduce the impact, I would always return an Ok response if the request is valid.